hibernate-core/src/main/java/org/hibernate/type/ArrayType.java

```diff
@@ -31,11 +34,20 @@
 
 	private final Class elementClass;
 	private final Class arrayClass;
+	private final Integer sqlType;
 
 	public ArrayType(TypeFactory.TypeScope typeScope, String role, String propertyRef, Class elementClass) {
 		super( typeScope, role, propertyRef );
 		this.elementClass = elementClass;
 		arrayClass = Array.newInstance(elementClass, 0).getClass();
+		sqlType = null;
+	}
+
+	public ArrayType(int sqlType, Class elementClass) {
+		super( null, null, null );
+		this.elementClass = elementClass;
+		arrayClass = Array.newInstance(elementClass, 0).getClass();
+		this.sqlType = sqlType;
```

Comment:
Can't say I'm excited about this. We basically have no idea if the `ArrayType` methods are working correctly without the super type fields being set.

The fact that you are basically overriding `nullSafeSet` in this case is also pretty concerning from a design point of view.

Don't know exactly what we should do but the current code seems to be a bit ad hoc.

Reply:
As long as a `sqlType` is provided, it seems fine to set the PS to an array of that Type. I ran all tests and they passed just fine.

The alternative is to create a new Type just for this particular use case.

Reply:
Yes, I'm wondering if it wouldn't be better to use a specific type with a very restricted scope.

Not really sure about it tbh. Maybe let's see how the array of arrays thing goes and then we can decide.